ts/src/header-validator/source.test.ts

```diff
@@ -2687,150 +2687,158 @@ const testCases: TestCase[] = [
+        path: ['attribution_scopes', 'limit'],
         msg: 'must be in the range [1, 4294967295]',
       },
+      {
+        path: ['attribution_scopes', 'values'],
+        msg: 'cannot be fully validated without a valid limit',
+      },
     ],
   },
   {
     name: 'attribution-scope-limit-not-integer',
     input: `{
       "destination": "https://a.test",
-      "attribution_scope_limit": 1.5
+      "attribution_scopes": {
+        "limit": 1.5,
+        "values": ["1"]
+      }
     }`,
     parseScopes: true,
     expectedErrors: [
       {
-        path: ['attribution_scope_limit'],
+        path: ['attribution_scopes', 'limit'],
         msg: 'must be an integer',
       },
-    ],
-  },
-  {
-    name: 'attribution-scope-limit-exceeds-max',
-    input: `{
-      "destination": "https://a.test",
-      "attribution_scope_limit": 4294967296
-    }`,
-    parseScopes: true,
-    expectedErrors: [
       {
-        path: ['attribution_scope_limit'],
-        msg: 'must be in the range [1, 4294967295]',
+        path: ['attribution_scopes', 'values'],
+        msg: 'cannot be fully validated without a valid limit',
       },
     ],
   },
   {
-    name: 'empty-attribution-scopes-with-limit',
+    name: 'attribution-scope-limit-exceeds-max',
     input: `{
       "destination": "https://a.test",
-      "attribution_scope_limit": 3,
-      "attribution_scopes": []
+      "attribution_scopes": {
+        "limit": 4294967296,
+        "values": ["1"]
+      }
     }`,
     parseScopes: true,
     expectedErrors: [
       {
-        path: ['attribution_scopes'],
-        msg: 'must be non-empty if attribution_scope_limit is set',
+        path: ['attribution_scopes', 'limit'],
+        msg: 'must be in the range [1, 4294967295]',
       },
-    ],
-  },
-  {
-    name: 'missing-attribution-scope-limit-attribution-scopes',
-    input: `{
-      "destination": "https://a.test",
-      "attribution_scopes": ["1", "2"]
-    }`,
-    parseScopes: true,
-    expectedErrors: [
       {
-        path: ['attribution_scopes'],
-        msg: 'must be empty if attribution_scope_limit is not set',
+        path: ['attribution_scopes', 'values'],
+        msg: 'cannot be fully validated without a valid limit',
       },
     ],
   },
   {
-    name: 'missing-attribution-scope-limit-max-event-states',
+    name: 'empty-attribution-scopes-with-limit',
     input: `{
       "destination": "https://a.test",
-      "max_event_states": 5
+      "attribution_scopes": {
+        "limit": 3,
+        "values": []
+      }
     }`,
     parseScopes: true,
     expectedErrors: [
       {
-        path: ['max_event_states'],
-        msg: 'must be default (3) if attribution_scope_limit is not set',
+        path: ['attribution_scopes', 'values'],
+        msg: 'must be non-empty if limit is set',
       },
     ],
   },
   {
-    name: 'invalid-attribution-scope-limit-attribution-scopes',
+    name: 'missing-attribution-scope-limit',
```

Comment:
1. Please also add a test case for a missing `values` field.
2. Please also add a test case for the `attribution_scopes` field not being a dictionary.

Reply:
Nice catch. Done.